### PR TITLE
Added a GetUnsigned function for getting unsigned values.

### PR DESCRIPTION
**Pull request by jcormier**
Allow getting unsigned integers without losing precision.

### CODE REVIEW

File: cpp/INIReader.cpp [hunk 1]
```diff
@@ -56,6 +56,16 @@ long INIReader::GetInteger(const string& section, const string& name, long defau
     return end > value ? n : default_value;
 }
 
+unsigned long INIReader::GetUnsignedInteger(const string& section, const string& name, unsigned long default_value) const
```
Comment (benhoyt) [inline cpp/INIReader.cpp:59]: Let's call this `GetUnsigned` for brevity (like with C / C++ itself, the "integer" is kind of obvious).
```suggestion
unsigned long INIReader::GetUnsigned(const string& section, const string& name, unsigned long default_value) const
```

File: cpp/INIReader.h [hunk 2]
```diff
@@ -66,6 +66,10 @@ class INIReader
     // not found or not a valid integer (decimal "1234", "-1234", or hex "0x4d2").
     INI_API long GetInteger(const std::string& section, const std::string& name, long default_value) const;
 
+    // Get an unsigned integer (unsigned long) value from INI file, returning default_value if
+    // not found or not a valid integer (decimal "1234", or hex "0x4d2").
```
Comment (benhoyt) [inline cpp/INIReader.h:70]: Let's wrap these comment lines at <80 chars, so after "returning" and after "or".

**Review by benhoyt — CHANGES_REQUESTED** [review]
Thanks! I'm good with adding this, though let's call it `GetUnsigned` as mentioned. Also, as you can see by the failed test output, you need to update `examples/cpptest.txt` so the C++ test passes.

**Comment by benhoyt** [comment]
Also, please update the PR title, and add a PR description (ideally with your use case).

**Comment by jcormier** [comment]
Thanks for the feedback. Working on it from my phone 😆. We shall see how that goes. I'll post another update when im done.

**Comment by jcormier** [comment]
Alright changes made. I'd like to squash all these commits together but can't seem to from the phone. So I'll get back to it next week when im back at my computer. Unless your dive with it as is

File: cpp/INIReader.h [hunk 3]
```diff
@@ -66,6 +66,11 @@ class INIReader
     // not found or not a valid integer (decimal "1234", "-1234", or hex "0x4d2").
     INI_API long GetInteger(const std::string& section, const std::string& name, long default_value) const;
 
+    // Get an unsigned integer (unsigned long) value from INI file,
+    // returning default_value if not found or not a valid integer 
+    // (decimal "1234", or hex "0x4d2").
+    INI_API unsigned long GetUnsignedInteger(const std::string& section, const std::string& name, unsigned long default_value) const;
```
Comment (benhoyt) [inline cpp/INIReader.h:72]:
```suggestion
    INI_API unsigned long GetUnsigned(const std::string& section, const std::string& name, unsigned long default_value) const;
```

**Comment by benhoyt** [comment]
Thanks. Two further changes required: 1) change `GetUnsignedInteger` to `GetUnsigned` in the header file too, and 2) update `examples/cpptest.txt` so the C++ test passes.

**Comment by jcormier** [comment]
Updated and squashed commits

**Comment by benhoyt** [comment]
Thanks!

**Pull request merged**